File: stacks/rails/app/controllers/api/staff_controller.rb
# スタッフ一覧（spec/openapi.yaml `/api/staff`）。password_hash は返さない
# （Staff#as_json で除外済み。spec/screens.md #21）。
module Api
  class StaffController < ApiController
    def index
      scope = Staff.all
      if params[:is_active].present?
        active = ActiveModel::Type::Boolean.new.cast(params[:is_active])
        scope = scope.where(is_active: active)
      end
      staffs = scope.order(:staff_code)
      render json: { items: staffs.map { |s| staff_json(s) }, total: staffs.size }
    end

    private

    def staff_json(s)
      { id: s.id, staff_code: s.staff_code, name: s.name, role: s.role, is_active: s.is_active }
    end
  end
end

File: stacks/laravel/app/Http/Controllers/Api/StaffController.php
<?php

namespace App\Http\Controllers\Api;

use App\Http\Controllers\Controller;
use App\Models\Staff;
use Illuminate\Http\JsonResponse;
use Illuminate\Http\Request;

/** スタッフ一覧のAPI。契約は spec/openapi.yaml `/api/staff`。 */
class StaffController extends Controller
{
    public function index(Request $request): JsonResponse
    {
        $query = Staff::query();
        if ($request->has('is_active')) {
            $query->where('is_active', $request->boolean('is_active'));
        }

        $items = $query->orderBy('id')->get();

        return response()->json([
            'items' => $items->map(fn ($s) => [
                'id' => $s->id,
                'staff_code' => $s->staff_code,
                'name' => $s->name,
                'role' => $s->role,
                'is_active' => (bool) $s->is_active,
            ])->values(),
            'total' => $items->count(),
        ]);
    }
}
